### DISCUSSION TITLE
Installing UiAutomator2 driver in appium returns Encountered an error when installing package: npm command 'install --save-dev --no-progress --no-audit --omit=peer --save-exact --global-style --no-package-lock appium-uiautomator2-driver --json' failed with code null. error

### DISCUSSION THREAD

**Discussion by Sudhakar1930** [appium/appium#20302] [Q&A]
Hi: I have the following versions of node, npm and appium. When I try to execute the script I got the error as 
Error: Could not find a driver for automationName 'UiAutomator2' and platformName 'ANDROID'. Have you installed a driver that supports those capabilities? Run 'appium driver list --installed' to see. (Lower-level error: Could not find installed driver to support given caps)

versions
---------
C:\Users\Rsoft>node -v
v20.14.0

C:\Users\Rsoft>npm -v
10.7.0

C:\Users\Rsoft>appium -v
2.9.0

C:\Users\Rsoft>node -v
v20.15.0

When I install UiAutomator2 driver , I get this below error.  Please let me know what I'm missing here...

C:\Users\Rsoft>appium driver install uiautomator2
√ Checking if 'appium-uiautomator2-driver' is compatible
× Installing 'uiautomator2' using NPM install spec 'appium-uiautomator2-driver'
Error: × Encountered an error when installing package: npm command 'install --save-dev --no-progress --no-audit --omit=peer --save-exact --global-style --no-package-lock appium-uiautomator2-driver --json' failed with code null.

STDOUT:

STDERR:

**Comment by mykola-mokhnach** [comment]
Upgrade you server to the most recent version
